extensions/ql-vscode/src/skeleton-query-wizard.ts

```diff
@@ -302,4 +280,24 @@ export class SkeletonQueryWizard {
     }
     return dbs[0];
   }
+
+  private async findExistingDatabaseItem() {
+    if (this.language === undefined) {
+      throw new Error("Language is undefined");
+    }
+
+    const databaseNwo = QUERY_LANGUAGE_TO_DATABASE_REPO[this.language];
+
+    return (
+      (await this.findDatabaseItemByNwo(
+        this.language,
+        databaseNwo,
+        this.databaseManager.databaseItems,
+      )) ||
+      (await this.findDatabaseItemByLanguage(
+        this.language,
+        this.databaseManager.databaseItems,
+      ))
+    );
```

Comment:
```suggestion
    const defaultDatabaseNwo = QUERY_LANGUAGE_TO_DATABASE_REPO[this.language];
    const defaultDatabaseItem = await this.findDatabaseItemByNwo(
      this.language,
      defaultDatabaseNwo,
      this.databaseManager.databaseItems,
    );
    if (defaultDatabaseItem !== undefined) {
      return defaultDatabaseItem;
    }

    return await this.findDatabaseItemByLanguage(
      this.language,
      this.databaseManager.databaseItems,
    );
```
I want to get your opinion here. I'm worried that the `||` expression where each side is an awaited promise is a bit unclear and could be misunderstood. This uses short-cut boolean logic, so if the left side returns something then the right side won't be executed at all, which is what we want, but at first glance it's hard to see the `||` lurking in there.

I think this proposed suggestion has identical behaviour but is just a bit more long-winded and readable at first glance. What do you think?

I also renamed the `databaseNwo` variable, but feel free to suggest any other name for this.

Reply:
Nice.